src/main/java/org/myrobotlab/service/ServoMixer.java

```diff
@@ -175,305 +256,345 @@ public Pose loadPose(String name) {
     try {
+      ServoMixerConfig c = (ServoMixerConfig)config;
+      
       if (filename == null) {
-        error("save sequence file name cannot be null");
+        error("save gesture file name cannot be null");
         return;
       }
 
-      if (json == null) {
-        error("sequence json cannot be null");
+      if (gesture == null) {
+        error("gesture json cannot be null");
         return;
       }
 
-      if (!filename.toLowerCase().endsWith(".seq")) {
-        filename += ".seq";
+      if (!filename.toLowerCase().endsWith(".yml")) {
+        filename += ".yml";
       }
 
-      Sequence seq = CodecUtils.fromJson(json, Sequence.class);
-      if (seq != null) {
-        String path = servoMixerDirectory + fs + filename;
+      // Gesture seq = CodecUtils.fromJson(json, Gesture.class);
+      
+      if (gesture != null) {
+        String path = c.gesturesDir + fs + filename;
         FileOutputStream fos = new FileOutputStream(path);
-        fos.write(CodecUtils.toPrettyJson(seq).getBytes());
+        fos.write(CodecUtils.toYaml(gesture).getBytes());
         fos.close();
-      }
-      invoke("getSequenceFiles");
+      } 
+      invoke("getGestureFiles");
     } catch (Exception e) {
       error(e);
     }
   }
+  
+  /**
+   * Save a {name}.yml file to the current poses directory.
+   * 
+   * @param name
+   *          name to save pose as
+   * @throws IOException
+   *           boom
+   * 
+   */
+  public void savePose(String name) throws IOException {
+    savePose(name, allServos);
+  }
+  
+  public void savePose(String name, Set<String> servos) throws IOException {
+    ServoMixerConfig c = (ServoMixerConfig)config;
 
-  @Override
-  public void startService() {
-    try {
-      List<String> all = Runtime.getServiceNamesFromInterface(ServoControl.class);
-      for (String sc : all) {
-        attach(sc);
-      }
-
-      File poseDirectory = new File(servoMixerDirectory);
-      if (!poseDirectory.exists()) {
-        poseDirectory.mkdirs();
+    if (servos == null) {
+      log.error("cannot save %s null servos");
+      return;
+    }
+        
+    Pose pose = new Pose();
+    
+    for (String servo : servos) {
+      ServoControl sc = (ServoControl)Runtime.getService(servo);
+      if (sc == null) {
+        error("servo %s null", name);
+        continue;
+      } else {
+        pose.moves.put(CodecUtils.shortName(servo), new PoseMove(sc.getCurrentInputPos(), sc.getSpeed()));
       }
-      super.startService();
-    } catch (Exception e) {
-      error(e);
     }
+
+    log.info("saving pose name {}", name);
+
+    String filename = c.posesDir + File.separator + name + ".yml";
+    FileIO.toFile(filename, CodecUtils.toYaml(pose).getBytes());
+    invoke("getPoseFiles");
+  }
+  
+  /**
+   * publishing a search to allow UI searching based on other
+   * service selections
+   * @param servos
+   * @return
+   */
+  public String search(String servos) {

```

Comment:
Not sure why this is needed, it only ever seems to be invoked from `onSelected` so couldn't the GUI just use whatever comes from `publishSelected()`?

Reply:
publishSelectedPath comes from JMonkeyEngine
so JMonkeyEngine -> ServoMixer -> ServoMixerGui